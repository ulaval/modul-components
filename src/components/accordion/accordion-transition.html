<transition name="m--is"
            @before-enter="beforeEnter"
            @enter="enter"
            @after-enter="afterEnter"
            @before-leave="beforeLeave"
            @after-leave="afterLeave">
    <slot></slot>
</transition>
