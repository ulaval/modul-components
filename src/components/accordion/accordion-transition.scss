.accordion-enter {
    height: 0;
}

.accordion-enter-active,
.accordion-leave-active {
    transition: height 0.3s ease-out;
    overflow: hidden;
}

.accordion-leave-to {
    height: 0 !important;
}
