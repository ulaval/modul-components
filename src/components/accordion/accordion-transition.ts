import Vue, { PluginObject, VNode, VNodeData, VueConstructor } from 'vue';

import { ACCORDION_TRANSITION_NAME } from '../component-names';

interface MAccordionTransitionProps {
    heightDelta?: number;
}

export const MAccordionTransition: VueConstructor<Vue> = Vue.extend({
    functional: true,
    render(createElement, context): VNode {
        let data: VNodeData = {
            props: {
                name: 'accordion'
            },
            on: {
                enter(el: HTMLElement): void {
                    const props: MAccordionTransitionProps = context.props as MAccordionTransitionProps;
                    const heightDelta = props.heightDelta
                        ? props.heightDelta
                        : 0;

                    el.style.height = el.scrollHeight - heightDelta + 'px';
                },
                afterEnter(el: HTMLElement): void {
                    el.style.removeProperty('height');
                },
                beforeLeave(el: HTMLElement): void {
                    el.style.height = el.scrollHeight + 'px';
                },
                afterLeave(el: HTMLElement): void {
                    el.style.removeProperty('height');
                }
            }
        };
        return createElement('transition', data, context.children);
    }
});

const AccordionTransitionPlugin: PluginObject<any> = {
    install(v, options): void {
        v.component(ACCORDION_TRANSITION_NAME, MAccordionTransition);
    }
};

export default AccordionTransitionPlugin;
