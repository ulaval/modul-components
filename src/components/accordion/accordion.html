<article class="m-accordion" :class="['m--is-' + propSkin, {'m--is-closed': !propOpen, 'm--is-disabled': propDisabled, 'm--has-icon-left': iconPosition != 'right', 'm--has-icon-large': iconSize=='large' }]">
    <div class="m-accordion__header"
         :class="{ 'm--has-padding': paddingHeader && padding }"
         tabindex="0"
         :id="propId"
         role="tab"
         :aria-expanded="propOpen"
         :aria-controls="propId + '-controls'"
         @keyup.self.enter="toggleAccordion"
         @click.prevent="toggleAccordion"
         ref="accordionHeader">
        <div class="m-accordion__header__content">
            <slot name="header"></slot>
            <m-i18n k="m-accordion:open" class="m-accordion__hidden" v-if="!propOpen"></m-i18n>
            <m-i18n k="m-accordion:close" class="m-accordion__hidden" v-if="propOpen"></m-i18n>
        </div>
        <span aria-hidden="true" class="m-accordion__header-icon" :class="{ 'm--is-closed': propOpen,
                        'm--is-left': iconPosition != 'right',
                        'm--is-large': iconSize=='large',
                        'm--has-border': propIconBorder }"></span>
    </div>
    <m-accordion-transition :height-delta="skin == 'default' || skin == 'dark' ? 24 : 0">
        <div class="m-accordion__body-wrap"
             role="tabpanel"
             :id="propId + '-controls'"
             :aria-labelledby="propId"
             v-if="propOpen">
            <div class="m-accordion__body" :class="{ 'm--has-padding': paddingBody && padding }">
                <slot></slot>
            </div>
        </div>
    </m-accordion-transition>
</article>
