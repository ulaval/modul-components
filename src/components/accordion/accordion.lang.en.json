{
    "m-accordion": {
        "open": "Click to display content",
        "close": "Click to hide content"
    }
}
