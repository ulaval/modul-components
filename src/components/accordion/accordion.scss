@import '../../styles/abstracts/variables';
$m-accordion--icon-l-width: 12px;
$m-accordion--icon-width: 10px;
$m-accordion--icon-l-height: 2px;
$m-accordion--icon-height: 1px;
$m-accordion--border-size: $m-border-width;
$m-accordion--border-color: $m-color--border;
$m-accordion--color: $m-color--accent;

.m-accordion {
    &.m--is-regular,
    &.m--is-light {
        > .m-accordion__header__content {
            &,
            h1,
            h2,
            h3,
            h4,
            h5,
            h6 {
                margin: 0;
                font-size: $m-font-size--l;
                font-weight: $m-font-weight--bold;
            }
        }
    }

    &.m--is-regular {
        margin-top: - $m-accordion--border-size;

        > .m-accordion {
            &__header,
            &__body-wrap > .m-accordion__body {
                padding: $m-padding $m-padding $m-padding 20px; // Magic number
            }

            &__header {
                position: relative;
                min-height: 56px;
                border-top: $m-accordion--border-size solid $m-accordion--border-color;
                border-bottom: $m-accordion--border-size solid $m-accordion--border-color;

                &::after {
                    position: absolute;
                    top: -$m-accordion--border-size;
                    bottom: -$m-accordion--border-size;
                    left: 0;
                    content: '';
                    width: $m-border-width--l;
                    background: $m-accordion--color;
                }
            }

            &__body-wrap {
                &.v-enter-to {
                    .m-accordion__body {
                        transform: translate3d(0, 0, 0);
                        opacity: 1;
                    }
                }

                &.v-enter,
                &.v-leave-to {
                    .m-accordion__body {
                        transform: translate3d(0, $m-margin--l, 0);
                        opacity: 0;
                    }
                }

                > .m-accordion__body {
                    transition: opacity $m-transition-duration--l ease, transform $m-transition-duration ease;
                }
            }
        }
    }

    &.m--is-light {
        margin: $m-margin--s 0;

        > .m-accordion {
            &__header {
                &:hover {
                    .m-accordion__header__content {
                        transition: color $m-transition-duration ease;
                        color: $m-color--interactive;
                    }
                }
            }

            &__body-wrap {
                > .m-accordion__body {
                    padding-left: $m-margin--l;
                    padding-top: $m-margin--s;
                    padding-bottom: $m-margin;
                }
            }
        }
    }

    &__header {
        display: flex;
        align-items: center;
        cursor: pointer;

        &:hover,
        &:active {
            outline: none;
        }

        &.m--is-icon-left {
            flex-direction: row-reverse;
        }

        &:not(.m--is-plain) {
            color: $m-color--text;
            text-decoration: none;
        }

        &__content {
            flex: 1;
        }

        &-icon {
            position: relative;
            display: block;

            &::before,
            &::after {
                position: absolute;
                top: 50%;
                left: 50%;
                content: '';
                background: $m-color--interactive;
                transform: translate3d(-50%, -50%, 0);
                transition: transform $m-transition-duration ease;
            }

            &::after {
                transition: transform $m-transition-duration ease, opacity 0.6s ease;
            }

            &.m--is-closed {
                &::before,
                &::after {
                    transform: translate3d(-50%, -50%, 0) rotate(90deg);
                }

                &::before {
                    opacity: 0;
                }
            }

            &:not(.m--is-large) {
                width: $m-accordion--icon-width;
                height: $m-accordion--icon-width;

                &::before {
                    width: $m-accordion--icon-width;
                    height: $m-accordion--icon-height;
                }

                &::after {
                    width: $m-accordion--icon-height;
                    height: $m-accordion--icon-width;
                }

                &.m--has-border {
                    width: $m-accordion--icon-width + 8;
                    height: $m-accordion--icon-width + 8;
                    border-width: $m-accordion--icon-height;
                }
            }

            &.m--is-large {
                width: $m-accordion--icon-l-width;
                height: $m-accordion--icon-l-width;

                &::before {
                    width: $m-accordion--icon-l-width;
                    height: $m-accordion--icon-l-height;
                }

                &::after {
                    width: $m-accordion--icon-l-height;
                    height: $m-accordion--icon-l-width;
                }

                &.m--has-border {
                    width: $m-accordion--icon-l-width + 8;
                    height: $m-accordion--icon-l-width + 8;
                    border-width: $m-accordion--icon-l-height;
                }
            }

            &.m--has-border {
                border-color: $m-color--interactive;
                border-style: solid;
            }

            &:not(.m--is-left) {
                margin-left: $m-margin;
            }

            &.m--is-left {
                margin-right: $m-margin--s;
            }
        }
    }

    &__hidden {
        @include m-visually-hidden();
    }
}
