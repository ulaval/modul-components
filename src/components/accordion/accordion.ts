import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';

import uuid from '../../utils/uuid/uuid';
import { ACCORDION_NAME } from '../component-names';
import I18nPlugin from '../i18n/i18n';
import AccordionTransitionPlugin from './accordion-transition';
import WithRender from './accordion.html?style=./accordion.scss';

export enum MAccordionSkin {
    Default = 'default',
    Dark = 'dark',
    Light = 'light',
    Plain = 'plain'
}

export enum MAccordionIconPosition {
    Left = 'left',
    Right = 'right'
}

export enum MAccordionIconSize {
    Small = 'small',
    Large = 'large'
}

export interface AccordionGateway extends Vue {
    propId: string;
    propOpen: boolean;
    propDisabled: boolean;
}

export interface AccordionGroupGateway {
    skin: MAccordionSkin;
    disabled: boolean;
    concurrent: boolean;
    addAccordion(accordion: AccordionGateway): void;
    removeAccordion(id: string): void;
    closeAllAccordions();
}

function isAccordionGroup(parent: any): parent is AccordionGroupGateway {
    return parent && 'addAccordion' in parent;
}

@WithRender
@Component
export class MAccordion extends Vue implements AccordionGateway {
    @Prop()
    public id?: string;

    @Prop({
        default: false
    })
    public open: boolean;

    @Prop({
        default: false
    })
    public disabled: boolean;

    @Prop({
        default: MAccordionSkin.Default,
        validator: value =>
            value === MAccordionSkin.Default ||
            value === MAccordionSkin.Dark ||
            value === MAccordionSkin.Light ||
            value === MAccordionSkin.Plain
    })
    public skin: MAccordionSkin;

    @Prop({
        default: MAccordionIconPosition.Left,
        validator: value =>
            value === MAccordionIconPosition.Left ||
            value === MAccordionIconPosition.Right
    })
    public iconPosition?: MAccordionIconPosition;

    @Prop()
    public iconBorder?: boolean;

    @Prop({
        default: MAccordionIconSize.Large,
        validator: value =>
            value === MAccordionIconSize.Small ||
            value === MAccordionIconSize.Large
    })
    public iconSize?: MAccordionIconSize;

    @Prop({ default: true })
    public padding: boolean;
    @Prop({ default: true })
    public paddingHeader: boolean;
    @Prop({ default: true })
    public paddingBody: boolean;

    $refs: {
        accordionHeader: HTMLElement;
    };

    private uuid: string = uuid.generate();
    private internalPropOpen: boolean = false;

    public get propId(): string {
        return this.id || this.uuid;
    }

    public get propOpen(): boolean {
        return this.internalPropOpen;
    }

    public set propOpen(value) {
        if (value !== this.internalPropOpen) {
            this.internalPropOpen = value;
            this.$emit('update:open', value);
        }
    }

    public get propDisabled(): boolean {
        return (isAccordionGroup(this.$parent) && this.$parent.disabled) ||
            this.disabled;
    }

    private created(): void {
        this.internalPropOpen = this.open;

        if (isAccordionGroup(this.$parent)) {
            this.$parent.addAccordion(this);
        }
    }

    private beforeDestroy(): void {
        if (isAccordionGroup(this.$parent)) {
            this.$parent.removeAccordion(this.propId);
        }
    }

    private get propSkin(): MAccordionSkin {
        return isAccordionGroup(this.$parent) ? this.$parent.skin : this.skin;
    }

    private get propIconBorder(): boolean {
        if (this.iconBorder) {
            return this.iconBorder;
        }

        return this.propSkin === MAccordionSkin.Light ? true : false;
    }

    private toggleAccordion(): void {
        if (!this.propDisabled) {
            const initialState = this.internalPropOpen;

            if (
                !this.internalPropOpen &&
                isAccordionGroup(this.$parent) &&
                this.$parent.concurrent
            ) {
                this.$parent.closeAllAccordions();
            }

            this.$refs.accordionHeader.blur();

            this.propOpen = !initialState;
            this.$emit('click', this.internalPropOpen);
        }
    }

    @Watch('open')
    private syncOpenProp(val: boolean): void {
        this.internalPropOpen = val;
    }
}

const AccordionPlugin: PluginObject<any> = {
    install(v, options): void {
        v.use(I18nPlugin);
        v.use(AccordionTransitionPlugin);
        v.component(ACCORDION_NAME, MAccordion);
    }
};

export default AccordionPlugin;
