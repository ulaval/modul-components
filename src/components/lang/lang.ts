import Vue from 'vue';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './lang.html';
import { I18N_NAME } from '../component-names';
import { ModulVue } from '../../utils/vue';

@WithRender
@Component
export class MLang extends ModulVue {
    public text: string = '';

    @Prop()
    public msg: string;

    public mounted() {
        this.text = this.$i18n.translate(this.msg);
    }
}

const LangPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(I18N_NAME, MLang);
    }
};

export default LangPlugin;
