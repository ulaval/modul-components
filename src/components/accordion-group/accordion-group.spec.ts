import '../../utils/polyfills';

import { mount, Slots } from '@vue/test-utils';
import Vue from 'vue';
import uuid from '../../utils/uuid/uuid';

import { addMessages } from '../../../tests/helpers/lang';
import { renderComponent } from '../../../tests/helpers/render';
import { MAccordion, MAccordionSkin } from '../accordion/accordion';
import AccordionGroupPlugin, { MAccordionGroup } from './accordion-group';

jest.mock('../../utils/uuid/uuid');
(uuid.generate as jest.Mock).mockReturnValue('uuid');

describe('MAcordionGroup', () => {
    beforeEach(() => {
        Vue.use(AccordionGroupPlugin);
        addMessages(Vue, [
            'components/accordion/accordion.lang.en.json',
            'components/accordion-group/accordion-group.lang.en.json'
        ]);
    });

    it('should render correctly', () => {
        const acn = mountGroup();

        return expect(renderComponent(acn.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with title', () => {
        const acn = mountGroup(
            {},
            {
                title: 'title'
            }
        );

        return expect(renderComponent(acn.vm)).resolves.toMatchSnapshot();
    });

    it('should not render toggle links when concurrent', () => {
        const acn = mountGroup({
            concurrent: true
        });

        return expect(renderComponent(acn.vm)).resolves.toMatchSnapshot();
    });

    it('should not render open all link when all accordions are closed', () => {
        const acn = mountGroup();

        return expect(renderComponent(acn.vm)).resolves.toMatchSnapshot();
    });

    it('should not render close all link when all accordions are opened', () => {
        const acn = mount(MAccordionGroup, {
            slots: {
                default: `<m-accordion :open="true">
                                <span slot="header">A</span>
                            </m-accordion>
                            <m-accordion :open="true">
                                <span slot="header">B</span>
                            </m-accordion>'`
            }
        });

        return expect(renderComponent(acn.vm)).resolves.toMatchSnapshot();
    });

    it('should not render both close all and open all link when state is mixed', () => {
        const acn = mount(MAccordionGroup, {
            slots: {
                default: `<m-accordion :open="true">
                                <span slot="header">A</span>
                            </m-accordion>
                            <m-accordion>
                                <span slot="header">B</span>
                            </m-accordion>'`
            }
        });

        return expect(renderComponent(acn.vm)).resolves.toMatchSnapshot();
    });

    it('should open all accordions when open all is clicked', () => {
        const acn = mountGroup();
        acn.update();

        acn.find('.m-accordion-group__header a').trigger('click');

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        expect(acrds.length).toBeGreaterThan(0);
        for (let i = 0; i < acrds.length; ++i) {
            expect(acrds.at(i).vm.propOpen).toBeTruthy();
        }
    });

    it('should close all accordions when close all is clicked', () => {
        const acn = mount(MAccordionGroup, {
            slots: {
                default: `<m-accordion id="a" :open="true">
                                <span slot="header">A</span>
                            </m-accordion>
                            <m-accordion id="b" :open="true">
                                <span slot="header">B</span>
                            </m-accordion>'`
            }
        });
        acn.update();

        acn.find('.m-accordion-group__header a').trigger('click');

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        expect(acrds.length).toBeGreaterThan(0);

        for (let i = 0; i < acrds.length; ++i) {
            expect(acrds.at(i).vm.propOpen).toBeFalsy();
        }
    });

    it('should should keep only one opened accordion when concurrent', () => {
        const acn = mountGroup({
            concurrent: true
        });
        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });

        acrds
            .at(0)
            .find('.m-accordion__header')
            .trigger('click');
        expect(acrds.at(0).vm.propOpen).toBeTruthy();
        expect(acrds.at(1).vm.propOpen).toBeFalsy();

        acrds
            .at(1)
            .find('.m-accordion__header')
            .trigger('click');
        expect(acrds.at(0).vm.propOpen).toBeFalsy();
        expect(acrds.at(1).vm.propOpen).toBeTruthy();
    });

    it('should cascade down skin to accordions', () => {
        const acn = mountGroup({
            skin: MAccordionSkin.Light
        });

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        expect(acrds.length).toBeGreaterThan(0);
        for (let i = 0; i < acrds.length; ++i) {
            expect(acrds.at(i).vm['propSkin']).toEqual(MAccordionSkin.Light);
        }
    });

    it('should cascade down disabled prop to accordions', () => {
        const acn = mountGroup({
            disabled: true
        });

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        expect(acrds.length).toBeGreaterThan(0);
        for (let i = 0; i < acrds.length; ++i) {
            expect(acrds.at(i).vm.propDisabled).toEqual(true);
        }
    });

    it('should use accordion disabled prop if group has none defined', () => {
        const acn = mount(MAccordionGroup, {
            slots: {
                default: `<m-accordion :disabled="true">
                                <span slot="header">A</span>
                            </m-accordion>
                            <m-accordion>
                                <span slot="header">B</span>
                            </m-accordion>'`
            }
        });

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        expect(acrds.at(0).vm.propDisabled).toEqual(true);
    });

    it('should open all accordions specified by id in openedIds prop', () => {
        const acn = mountGroup({
            openedIds: ['b']
        });

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        expect(acrds.at(0).vm.propOpen).toBeFalsy();
        expect(acrds.at(1).vm.propOpen).toBeTruthy();
    });

    it('should should sync openedIds prop when a accordion is closed or opened', () => {
        const acn = mountGroup({
            openedIds: []
        });

        const acrds = acn.findAll<MAccordion>({ name: 'MAccordion' });
        acrds
            .at(0)
            .find('.m-accordion__header')
            .trigger('click');

        expect(acn.emitted('update:openedIds')[0][0]).toEqual(['a']);
    });

    const mountGroup = (propsData?: object, slots?: Slots) => {
        return mount(MAccordionGroup, {
            propsData: propsData,
            slots: {
                default: `<m-accordion id="a">
                                <span slot="header">A</span>
                            </m-accordion>
                            <m-accordion id="b">
                                <span slot="header">B</span>
                            </m-accordion>'`,
                ...slots
            }
        });
    };
});
