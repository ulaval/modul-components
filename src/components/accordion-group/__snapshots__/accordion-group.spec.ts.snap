// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MAcordionGroup should not render both close all and open all link when state is mixed 1`] = `
<article class="m-accordion-group">
    <div class="m-accordion-group__header m--no-title">
        <div>
            <a href="#" class="m-link m--is-unvisited m--is-button"> <span class="m-link__text"><span class="m-i18n">Open all</span></span>
            </a>
        </div>
    </div>
    <div class="m-accordion-group__body">
        <article class="m-accordion m--is-default m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="uuid" role="tab" aria-expanded="true" aria-controls="uuid-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>A</span> <span class="m-i18n m-accordion__hidden">Click to hide content.</span></div> <span aria-hidden="true" class="m-accordion__header-icon m--is-closed m--is-left m--is-large"></span></div>
            <div role="tabpanel" id="uuid-controls"
                aria-labelledby="uuid" class="m-accordion__body-wrap">
                <div class="m-accordion__body m--has-padding"></div>
            </div>
        </article>
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="uuid" role="tab" aria-controls="uuid-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>B</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>'</div>
</article>
`;

exports[`MAcordionGroup should not render close all link when all accordions are opened 1`] = `
<article class="m-accordion-group">
    <div class="m-accordion-group__header m--no-title">
        <div>
            <a href="#" class="m-link m--is-unvisited m--is-button"> <span class="m-link__text"><span class="m-i18n">Close all</span></span>
            </a>
        </div>
    </div>
    <div class="m-accordion-group__body">
        <article class="m-accordion m--is-default m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="uuid" role="tab" aria-expanded="true" aria-controls="uuid-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>A</span> <span class="m-i18n m-accordion__hidden">Click to hide content.</span></div> <span aria-hidden="true" class="m-accordion__header-icon m--is-closed m--is-left m--is-large"></span></div>
            <div role="tabpanel" id="uuid-controls"
                aria-labelledby="uuid" class="m-accordion__body-wrap">
                <div class="m-accordion__body m--has-padding"></div>
            </div>
        </article>
        <article class="m-accordion m--is-default m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="uuid" role="tab" aria-expanded="true" aria-controls="uuid-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>B</span> <span class="m-i18n m-accordion__hidden">Click to hide content.</span></div> <span aria-hidden="true" class="m-accordion__header-icon m--is-closed m--is-left m--is-large"></span></div>
            <div role="tabpanel" id="uuid-controls"
                aria-labelledby="uuid" class="m-accordion__body-wrap">
                <div class="m-accordion__body m--has-padding"></div>
            </div>
        </article>'</div>
</article>
`;

exports[`MAcordionGroup should not render open all link when all accordions are closed 1`] = `
<article class="m-accordion-group">
    <div class="m-accordion-group__header m--no-title">
        <div>
            <a href="#" class="m-link m--is-unvisited m--is-button"> <span class="m-link__text"><span class="m-i18n">Open all</span></span>
            </a>
        </div>
    </div>
    <div class="m-accordion-group__body">
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="a" role="tab" aria-controls="a-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>A</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="b" role="tab" aria-controls="b-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>B</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>'</div>
</article>
`;

exports[`MAcordionGroup should not render toggle links when concurrent 1`] = `
<article class="m-accordion-group">
    <div class="m-accordion-group__body">
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="a" role="tab" aria-controls="a-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>A</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="b" role="tab" aria-controls="b-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>B</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>'</div>
</article>
`;

exports[`MAcordionGroup should render correctly 1`] = `
<article class="m-accordion-group">
    <div class="m-accordion-group__header m--no-title">
        <div>
            <a href="#" class="m-link m--is-unvisited m--is-button"> <span class="m-link__text"><span class="m-i18n">Open all</span></span>
            </a>
        </div>
    </div>
    <div class="m-accordion-group__body">
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="a" role="tab" aria-controls="a-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>A</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="b" role="tab" aria-controls="b-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>B</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>'</div>
</article>
`;

exports[`MAcordionGroup should render correctly with title 1`] = `
<article class="m-accordion-group">
    <div class="m-accordion-group__header">
        <div class="m-accordion-group__header__title">title</div>
        <div>
            <a href="#" class="m-link m--is-unvisited m--is-button"> <span class="m-link__text"><span class="m-i18n">Open all</span></span>
            </a>
        </div>
    </div>
    <div class="m-accordion-group__body">
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="a" role="tab" aria-controls="a-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>A</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>
        <article class="m-accordion m--is-default m--is-closed m--has-icon-left m--has-icon-large">
            <div tabindex="0" id="b" role="tab" aria-controls="b-controls" class="m-accordion__header m--has-padding">
                <div class="m-accordion__header__content"><span>B</span> <span class="m-i18n m-accordion__hidden">Click to display content.</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large"></span></div>
        </article>'</div>
</article>
`;
